lib/rubocop/cop/rails/redundant_receiver_in_with_options.rb

```diff
@@ -66,7 +70,8 @@ def autocorrect(node)
         private
 
         def block_argument_range(node)
-          block_argument = node.parent.parent.children[1].loc.expression
+          block_node = node.each_ancestor(:block).first
+          block_argument = block_node.children[1].loc.expression
```

Comment:
Instead of `.loc.expression`, I believe you abbreviate the chain by using  `.source_range`.